model/textparse/interface.go

```diff
@@ -78,28 +79,66 @@ type Parser interface {
 	Next() (Entry, error)
 }
 
-// New returns a new parser of the byte slice.
-//
-// This function always returns a valid parser, but might additionally
-// return an error if the content type cannot be parsed.
-func New(b []byte, contentType string, parseClassicHistograms, skipOMCTSeries bool, st *labels.SymbolTable) (Parser, error) {
+// Helper function to returns the mediaType of a required parser checking contentType
+// first and falling back to fallbackType if necessary.
+// As fallbackType comes from the parse `fallback_scrape_protocol` config option it
+// is assumed to be a valid/supported mediaType.
+func newHelper(contentType, fallbackType string) (string, error) {
 	if contentType == "" {
-		return NewPromParser(b, st), nil
+		if fallbackType == "" {
+			return "", errors.New("Non-compliant scraper sending blank Content-Type and no fallback_scrape_protocol specified for target")
+		}
+		// We don't set the error here so that we're kind and mirror v2 behaviour
+		return fallbackType, nil
 	}
 
+	// We have a contentType, parse it
 	mediaType, _, err := mime.ParseMediaType(contentType)
 	if err != nil {
-		return NewPromParser(b, st), err
+		if fallbackType == "" {
+			retErr := errors.New("Cannot parse Content-Type and no fallback_scrape_protocol for target")
+			return "", errors.Join(retErr, err)
+		}
+		retErr := errors.New("Could not parse received Content-Type, using fallback_scrape_protocol for target")
+		return fallbackType, errors.Join(retErr, err)
+	}
+
+	// We have a valid media type, either we recognise it and can use it
+	// or we have to error
+	switch mediaType {
+	case "application/openmetrics-text", "application/vnd.google.protobuf", "text/plain":
+		return mediaType, nil
+	}
+	// We're here because we have no regonised mediaType
+	if fallbackType == "" {
+		return "", errors.New("Scraper sending unrecognisable Content-Type and no fallback_scrape_protocol specified for target")
 	}
+	return fallbackType, errors.New("Content-Type not recognised mediaType, using fallback_scrape_protocol for target")
```

Comment:
Let's include the invalid content-type string in the error message.

Reply:
Done (and to the corresponding error above for no fallback) and fixed the associated tests.